src/main/java/frc/robot/systems/ClimberMechFSMLeft.java

```diff
@@ -0,0 +1,177 @@
+		motor = new CANSparkMax(HardwareMap.CAN_ID_SPARK_LEFT_CLIMBER_MOTOR,
+						CANSparkMax.MotorType.kBrushless);
+		motor.setIdleMode(CANSparkMax.IdleMode.kBrake);
+
+		peakLimitSwitch = motor.getReverseLimitSwitch(SparkLimitSwitch.Type.kNormallyClosed);
+		peakLimitSwitch.enableLimitSwitch(false);
+
+		// Reset state machine
+		reset();
+	}
+
+	/* ======================== Public methods ======================== */
+	/**
+	 * Return current FSM state.
+	 * @return Current FSM state
+	 */
+	public ClimberMechFSMState getCurrentState() {
+		return currentState;
+	}
+	/**
+	 * Reset this system to its start state. This may be called from mode init
+	 * when the robot is enabled.
+	 *
+	 * Note this is distinct from the one-time initialization in the constructor
+	 * as it may be called multiple times in a boot cycle,
+	 * Ex. if the robot is enabled, disabled, then reenabled.
+	 */
+	public void reset() {
+		currentState = ClimberMechFSMState.IDLE_STOP;
+
+		// Call one tick of update to ensure outputs reflect start state
+		update(null);
+	}
+
+	/**
+	 * Update FSM based on new inputs. This function only calls the FSM state
+	 * specific handlers.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 */
+	public void update(TeleopInput input) {
+
+		if (input == null) {
+			return;
+		}
+
+		if (peakLimitSwitch.isPressed()) {
+			limitPressed = true;
+		}
+
+		switch (currentState) {
+			case IDLE_STOP:
+				handleIdleState(input);
+				break;
+			case RETRACTING:
+				handleRetractingState(input);
+				break;
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+		SmartDashboard.putString("Current State", currentState.toString());
+		SmartDashboard.putBoolean("Bottom Limit Switch Pressed", peakLimitSwitchHit());
+		SmartDashboard.putBoolean("Retract Button Pressed", input.isRetractClimberButtonPressed());
+		currentState = nextState(input);
+	}
+
+	/**
+	 * Performs specific action based on the autoState passed in.
+	 * @param autoState autoState that the subsystem executes.
+	 * @return if the action carried out in this state has finished executing
+	 */
+	public boolean updateAutonomous(AutoFSMState autoState) {
+		switch (autoState) {
+			default:
+				return true;
+		}
+	}
+
+	/* ======================== Private methods ======================== */
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private ClimberMechFSMState nextState(TeleopInput input) {
+		if (input == null) {
+			return ClimberMechFSMState.IDLE_STOP;
+		}
```

Comment:
You can get rid of the above if statement, you're never gonna have the case where input is null in nextState()